src/constants/memory/hram.asm

```diff
@@ -45,291 +45,391 @@ hNeedsUpdatingBGTiles:: ; FF90
-hDMARoutine:: ; FFC0
-  ds $A
+hDMARoutine::
+  ds $A ; FFC0 - FFC9
 
-; Unlabeled
-hFFCA:: ; FFCA
-  ds 1
+; not used
+hFFCA:
+  ds 1 ; FFCA
 
-hPressedButtonsMask:: ; FFCB
-  ; The pressed buttons on the joypad.
-  ; During all the time a button is pressed, the matching bit on this variable is set.
-  ;
-  ; See J_* constants for values.
-  ds 1
+; The pressed buttons on the joypad.
+; During all the time a button is pressed, the matching bit on this variable is set.
+;
+; See J_* constants for values.
+hPressedButtonsMask::
+  ds 1 ; FFCB
 
-hJoypadState:: ; FFCC
-  ; The buttons newly pressed on the joypad.
-  ; The matching bit is set only during the first frame the button is pressed.
-  ; Afterwards, the bit is cleared.
-  ;
-  ; See J_* constants for values.
-  ds 1
+; The buttons newly pressed on the joypad.
+; The matching bit is set only during the first frame the button is pressed.
+; Afterwards, the bit is cleared.
+;
+; See J_* constants for values.
+hJoypadState::
+  ds 1 ; FFCC
 
-hSwordIntersectedAreaY:: ; FFCD
-  ; Topmost corner of the area intercepted by Link's sword
-  ; Vary from 00 to 80 by increments of 10
-  ds 1
+; Topmost corner of the area intercepted by Link's sword
+; Vary from 00 to 80 by increments of 10
+hSwordIntersectedAreaY::
+  ds 1 ; FFCD
 
-hSwordIntersectedAreaX:: ; FFCE
-  ; Leftmost corner of the area intercepted by Link's sword
-  ; Vary from 00 to A0 by increments of 10
-  ds 1
+; Leftmost corner of the area intercepted by Link's sword
+; Vary from 00 to A0 by increments of 10
+hSwordIntersectedAreaX::
+  ds 1 ; FFCE
 
 ; Unlabeled
-ds $2
-
-hNeedsRenderingFrame:: ; FFD1
- ds 1
-
-hFFD2:: ; FFD2
-  ds 5
-
-; Scratch values, used for many different uses
-
-hScratch0:: ; FFD7
-  ds 1
-hScratch1:: ; FFD8
-  ds 1
-hScratch2:: ; FFD9
-  ds 1
-hScratch3:: ; FFDA
-  ds 1
-hScratch4:: ; FFDB
-  ds 1
-hScratch5:: ; FFDC
-  ds 1
-hScratch6:: ; FFDD
-  ds 1
-hScratch7:: ; FFDE
-  ds 1
-hScratch8:: ; FFDF
-  ds 1
-hScratch9::          ; FFE0
-hBGMapOffsetHigh:: ; FFE0
-  ds 1
-hScratchA::          ; FFE1
-hBGMapOffsetLow::  ; FFE1
-  ds 1
-hScratchB:: ; FFE2
-  ds 1
-hScratchC:: ; FFE3
-  ds 1
-hScratchD:: ; FFE4
-  ds 1
-hScratchE:: ; FFE5
-  ds 1
-hScratchF:: ; FFE6
-hFreeWarpDataAddress:: ; FFE6
-  ; Address of the first free warp data slot
-  ds 1
+hFFCF::
+  ds $1 ; FFCF
 
-hFrameCounter:: ; FFE7
-  ; wraps around 00-FF
-  ds 1
+; Unlabeled
+hFFD0::
+  ds $1 ; FFD0
 
-; This location has different uses
-hFFE8::                   ; FFE8
-hRoomBank::             ; FFE8
-hDialogBackgroundTile:: ; FFE8
-  ds 1
+; boolean value if changes needs to be rendered
+; 1 = frame rendering is needed,
+; 0 = frame rendering was done
+hNeedsRenderingFrame::
+  ds 1 ; FFD1
 
-; This location has diffent uses
-hFFE9::                   ; FFE9
-hDungeonFloorTile::       ; FFE9
-hIndexOfObjectBelowLink:: ; FFE9
-  ds 1
+; Unlabeled
+hFFD2::
+  ds 5 ; FFD2
+
+; used in many different cases
+hMultiPurpose0::
```

Comment:
I see you renamed the generic purpose registers from `hScratch0` to `hMultiPurpose0`.

Is is because the "scratch" wording wasn't clear to you?

I initially chose "scratch" because it seems widely used in embedded systems programming (like [scratch registers](https://community.arm.com/developer/tools-software/tools/f/armds-forum/843/concept-of-scratch-and-preserved-registers) or [scratch memory](https://en.wikipedia.org/wiki/Scratchpad_memory)), so I guess I would rather keep it, unless for a good reason.

Reply:
Yes I renamed Scratch to MultiPurpose for readability.
I googeled for scratch, but all I found was stuff for the kiddy language scatch. You know the one with the cat as mascot and with drag and drop programming.
So I thought MultiPurpose is more specific and clear.
I made it a separate commit, so its easy to revert, if you want so.

Reply:
Yes, maybe if you googled and were still confused it means the name is not good enough. `MultiPurpose` seems good to me.